src/platform/Module.c

```diff
@@ -0,0 +1,373 @@
+            info->version.minor = json_object_get_number(object, INFO_VERSION_MINOR);
+            info->version.patch = json_object_get_number(object, INFO_VERSION_PATCH);
+            info->version.tweak = json_object_get_number(object, INFO_VERSION_TWEAK);
+
+            if ((lifetime < 0) || (2 < lifetime))
+            {
+                LOG_ERROR("Module info has invalid lifetime: %d", lifetime);
+                status = EINVAL;
+            }
+            else
+            {
+                info->lifetime = lifetime;
+            }
+
+            if ((NULL != licenseUri) && (NULL == (info->licenseUri = strdup(licenseUri))))
+            {
+                LOG_ERROR("Failed to allocate memory for module license URI");
+                status = ENOMEM;
+            }
+
+            if ((NULL != projectUri) && (NULL == (info->projectUri = strdup(projectUri))))
+            {
+                LOG_ERROR("Failed to allocate memory for module project URI");
+                status = ENOMEM;
+            }
+
+            if (NULL == (info->components = (char**)calloc(componentsCount, sizeof(char*))))
+            {
+                LOG_ERROR("Failed to allocate memory for module components");
+                status = ENOMEM;
+            }
+            else
+            {
+                info->componentCount = componentsCount;
+
+                for (int i = 0; i < componentsCount; i++)
+                {
+                    if (NULL == (component = (char*)json_array_get_string(components, i)))
+                    {
+                        LOG_ERROR("Failed to get component name at index: %d", i);
+                        status = EINVAL;
+                        break;
+                    }
+                    else if (NULL == (info->components[i] = strdup(component)))
+                    {
+                        LOG_ERROR("Failed to copy component name at index: %d", i);
+                        status = ENOMEM;
+                        break;
+                    }
+                }
+            }
+
+            if (0 == status)
+            {
+                *moduleInfo = info;
+            }
+            else
+            {
+                FreeModuleInfo(info);
+                info = NULL;
+            }
+        }
+    }
+
+    return status;
+}
+
+MODULE* LoadModule(const char* client, const char* path)
+{
+    int status = 0;
+    MODULE* module = NULL;
+    MODULE_INFO* info = NULL;
+    JSON_Value* value = NULL;
+    MMI_JSON_STRING payload = NULL;
+    int payloadSize = 0;
+
+    printf("Loading module %s", path);
+
+    if (NULL == client)
+    {
+        LOG_ERROR("Invalid (null) client");
+        status = EINVAL;
+    }
+    else if (NULL == path)
+    {
+        LOG_ERROR("Invalid (null) path");
+        status = EINVAL;
+    }
+    else if (NULL == (module = (MODULE*)calloc(1, sizeof(MODULE))))
+    {
+        LOG_ERROR("Failed to allocate memory for module");
```

Comment:
>LOG_ERROR [](http://example.com/codeflow?start=8&length=9)

Why these? Use the normal OSConfig logging macros instead directly. Get rid of the extra layer of macros that are really unnecessary.